### PR TITLE
[DC-2376] De-duplicate records in _mapping_person_src_hpos

### PR DESCRIPTION
**Pull request by hiro-mishima**
The reason why it is safe to de-duplicate `_mapping_person_src_hpos`:
- `_mapping_person_src_hpos` is referenced only here in the whole DEID process -> [link to the code](https://github.com/all-of-us/curation/blob/develop/data_steward/deid/config/ids/config.json#L39)
- In the SQL above, `_mapping_person_src_hpos` is used in a `SELECT DISTINCT` statement. This DISTINCT statement returns the same result with or without `_mapping_person_src_hpos` having duplicates.

Notes:
- `_mapping_person_src_hpos` is created to deid `observation` table -> [link to the code](https://github.com/all-of-us/curation/blob/develop/data_steward/deid/aou.py#L265)
- Also, see the comment in the JIRA ticket for what I did for debugging/testing.

### CODE REVIEW

File: data_steward/deid/aou.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def create_person_id_src_hpo_map(client, input_dataset):
         sql_statement.append(
             sql.format(table=table, input_dataset=input_dataset))
 
-    final_query = ' UNION ALL '.join(sql_statement)
+    final_query = ' UNION '.join(sql_statement)
```
Comment (lrwb-aou) [inline data_steward/deid/aou.py:164]: This might be better as ` UNION DISTINCT `.  Thoughts?

Comment (hiro-mishima) [reply data_steward/deid/aou.py:164]: I thought `UNION` and `UNION DISTINCT` were the same. They both remove the duplicate from the output.
Do they behave differently?

Comment (lrwb-aou) [reply data_steward/deid/aou.py:164]: This used to be ` UNION ALL `.  When I try the following in the GCP UI, it errors out.
```sql
select 'foo' as name
union
select 'bar' as name
union
select 'foo' as name;
```

I have to specify either ` UNION ALL ` or ` UNION DISTINCT `.

Comment (hiro-mishima) [reply data_steward/deid/aou.py:164]: Thanks for pointing it out! I did not know `UNION` does not work with BigQuery 😮 Let me update it.

**Review by lrwb-aou — APPROVED** [review]
This is a nice investigation, analysis, and optimization.  Looks good!

**Pull request merged**